android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUITimePicker.java

```diff
@@ -160,15 +174,36 @@ public void setValue(long value, boolean suppressEvent)
 	@Override
 	public void onTimeChanged(TimePicker view, int hourOfDay, int minute)
 	{
+		detectTimeChange();
+	}
+	
+	private void detectTimeChange() {
 		Calendar calendar = Calendar.getInstance();
-		calendar.set(Calendar.HOUR_OF_DAY, hourOfDay);
-		calendar.set(Calendar.MINUTE, minute);
-		if (!suppressChangeEvent) {
+		TimePicker picker = (TimePicker) getNativeView();
+		boolean hasChanged;
+		if (Build.VERSION.SDK_INT >= 23) {
+			hasChanged = lastSelectedHour != picker.getHour() || lastSelectMin != picker.getMinute();
```

Comment:
@garymathews, the "opcode" warnings are definitely happening. They can be seen via "adb logcat". Our exiting codebase has this issue everywhere and there is a desire to fix it per [TIMOB-20258]. The solution I posted above resolves it by taking advantage of Java's lazy class loading, which is the recommended solution by Google (link below) and is the same solution Google uses in their Android "support" libraries.
   https://android-developers.googleblog.com/2010/07/how-to-have-your-cupcake-and-eat-it-too.html